app/src/main/java/com/example/appharrypotter/fragment/ListaPersonagensFragment.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.appharrypotter.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.example.appharrypotter.adapter.ListaPersonagemAdapter
+import com.example.appharrypotter.databinding.FragmentListaPersonagensBinding
+import com.example.appharrypotter.viewModel.PersonagensViewModel
+
+class ListaPersonagensFragment : Fragment() {
+
+    private var _binding : FragmentListaPersonagensBinding? = null
+    private val binding get() = _binding!!
+    private val viewModel : PersonagensViewModel by viewModels()
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentListaPersonagensBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        iniciarRecycler()
+    }
+
+    private fun iniciarRecycler() {
+        viewModel.listaPersonagens.observe(viewLifecycleOwner) {
+            val adapter = ListaPersonagemAdapter()
```

Comment:
Criar o adapter dessa forma vai resultar em uma nova instância de adapter sempre que a lista de personagens for atualizada, o ideal é criá-lo como variavel global da classe utilizando o by lazy